packages/amplify_core/lib/src/types/datastore/hub/hub_event_element.dart

```diff
@@ -29,10 +29,7 @@ Model _parseModelFromMap(
 ) {
   final serializedElement = serializedHubEventElement['element'] as Map;
   final modelName = serializedHubEventElement['modelName'] as String;
-  final modelData = serializedElement['model'] as Map;
-  final serializedModelData =
-      (modelData['serializedData'] as Map).cast<String, dynamic>();
-  return provider
-      .getModelTypeByModelName(modelName)
-      .fromJson(serializedModelData);
+  final modelData =
+      Map<String, dynamic>.from(serializedElement['model'] as Map);
```

Comment:
Use `cast` to avoid allocating a new Map:

```suggestion
      (serializedElement['model'] as Map).cast<String, dynamic>();
```